src/main/java/com/adobe/aem/compgenerator/javacodemodel/ImplementationBuilder.java

```diff
@@ -253,7 +255,9 @@ private void addGetter(JDefinedClass jc, JFieldVar jFieldVar) {
 
 
         if (jFieldVar.type().erasure().fullName().equals(List.class.getName())) {
-            getMethod.body()._return(codeModel.ref(Collections.class).staticInvoke("unmodifiableList").arg(jFieldVar));
+            JConditional ifNull = getMethod.body()._if(JExpr.ref(jFieldVar.name()).ne(JExpr._null()));
+            ifNull._then()._return(codeModel.ref(Collections.class).staticInvoke("unmodifiableList").arg(jFieldVar));
+            ifNull._else()._return(JExpr._null());
```

Comment:
Is there no way to do this as a ternary by using `directStatement()`?  I know it's not the "cleanest" implementation here, but it would result in cleaner end code.

```
getMethod.body().directStatement("return " + jFieldVar.name() + " != null ? Collections.unmodifiableList(" + jFieldVar.name() + ") : null;");
```
worked nicely except for missing the import of `java.util.Collections`

Reply:
and assuming we *cant* do that, I think at least this would be a _little_ cleaner:
```
if (this.myvar != null) {
    return Collections.unmodifiableList(this.myvar);
}
return null;
```

Reply:
actually, just found this: `com.sun.codemodel.JOp#cond`

Looks like we can do a ternary!

Reply:
Done. It writes out ternary operation.